File: backend/schemas/documentation.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class DocumentationPageCreate(BaseModel):
    title: str = Field(min_length=2, max_length=180)
    content: str = Field(min_length=1)
    task_id: Optional[int] = None


class DocumentationPageUpdate(BaseModel):
    title: Optional[str] = Field(default=None, min_length=2, max_length=180)
    content: Optional[str] = Field(default=None, min_length=1)
    task_id: Optional[int] = None


class DocumentationPageOut(BaseModel):
    id: int
    project_id: int
    task_id: Optional[int] = None
    title: str
    content: str
    created_by_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/services/documentation.ts
import api from "@/lib/axios";

export interface DocumentationPage {
  id: number;
  project_id: number;
  task_id?: number | null;
  title: string;
  content: string;
  created_by_id?: number | null;
  created_at: string;
  updated_at: string;
}

export interface DocumentationRevision {
  id: number;
  page_id: number;
  project_id: number;
  task_id?: number | null;
  title: string;
  content: string;
  action: string;
  actor_id?: number | null;
  created_at: string;
}

export interface DocumentationPageCreate {
  title: string;
  content: string;
  task_id?: number | null;
}

export interface DocumentationPageUpdate {
  title?: string;
  content?: string;
  task_id?: number | null;
}

export const getProjectDocumentationPages = async (
  projectId: number
): Promise<DocumentationPage[]> => {
  const response = await api.get(`/documentation/project/${projectId}`);
  return response.data;
};

export const createDocumentationPage = async (
  projectId: number,
  data: DocumentationPageCreate
): Promise<DocumentationPage> => {
  const response = await api.post(`/documentation/project/${projectId}`, data);
  return response.data;
};

export const updateDocumentationPage = async (
  pageId: number,
  data: DocumentationPageUpdate
): Promise<DocumentationPage> => {
  const response = await api.put(`/documentation/${pageId}`, data);
  return response.data;
};

export const getDocumentationPageHistory = async (
  pageId: number
): Promise<DocumentationRevision[]> => {
  const response = await api.get(`/documentation/${pageId}/history`);
  return response.data;
};

export const deleteDocumentationPage = async (pageId: number): Promise<void> => {
  await api.delete(`/documentation/${pageId}`);
};

export const generateDocumentationFromTask = async (
  taskId: number
): Promise<DocumentationPage> => {
  const response = await api.post(`/documentation/from-task/${taskId}`);
  return response.data;
};
